nomad/structs/services.go

```diff
@@ -676,6 +677,10 @@ func (s *Service) Copy() *Service {
 	ns.CanaryMeta = maps.Clone(s.CanaryMeta)
 	ns.TaggedAddresses = maps.Clone(s.TaggedAddresses)
 
+	if s.Weights != nil {
+		ns.Weights = s.Weights.Copy()
+	}
```

Comment:
The `Copy` has a nil check, so we can safely skip this conditional.

```suggestion	
	ns.Weights = s.Weights.Copy()
```